snapd-glib/snapd-notice.c

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2 or version 3 of the License.
+ * See http://www.gnu.org/copyleft/lgpl.html the full text of the license.
+ */
+
+#include "snapd-notice.h"
+
+/**
+ * SECTION: snapd-notice
+ * @short_description: Notices element
+ * @include: snapd-glib/snapd-glib.h
+ *
+ * A #SnapdNotice contains information about a change that is notified.
+ */
+
+/**
+ * SnapdNotice:
+ *
+ * #SnapdNotice contains information on a notification element.
+ *
+ * Since: 1.5
+ */
+
+enum
+{
+    PROP_ID = 1,
+    PROP_USER_ID,
+    PROP_TYPE,
+    PROP_KEY,
+    PROP_FIRST_OCCURRED,
+    PROP_LAST_OCCURRED,
+    PROP_LAST_REPEATED,
+    PROP_OCCURRENCES,
+    PROP_LAST_DATA,
+    PROP_REPEAT_AFTER,
+    PROP_EXPIRE_AFTER,
+    PROP_LAST
+};
+
+struct _SnapdNotice
+{
+    GObject parent_instance;
+
+    gchar *id;
+    gchar *userid;
+    SnapdNoticeType type;
+    gchar *key;
+    GDateTime *first_occurred;
+    GDateTime *last_occurred;
+    GDateTime *last_repeated;
+    GHashTable *data;
+    gint occurrences;
+    GTimeSpan repeat_after;
+    GTimeSpan expire_after;
+};
+
+
+G_DEFINE_TYPE (SnapdNotice, snapd_notice, G_TYPE_OBJECT)
+
+/**
+ * snapd_notice_get_id:
+ * @notice: a #SnapdNotice.
+ *
+ * Get the unique ID for this change.
+ *
+ * Returns: (transfer none): an ID.
+ *
+ * Since: 1.5
+ */
+const gchar *
+snapd_notice_get_id (SnapdNotice *self)
+{
+    g_return_val_if_fail (SNAPD_IS_NOTICE (self), NULL);
+    return self->id;
+}
+
+/**
+ * snapd_notice_get_user_id:
+ * @notice: a #SnapdNotice.
+ *
+ * Get the user ID for this change, or NULL if no user is defuned
+ *
+ * Returns: (transfer none): an user ID.
+ *
+ * Since: 1.5
+ */
+const gchar *
+snapd_notice_get_user_id (SnapdNotice *self)
+{
+    g_return_val_if_fail (SNAPD_IS_NOTICE (self), NULL);
+    return self->userid;
+}
+
+/**
+ * snapd_notice_get_notice_type:
+ * @notice: a #SnapdNotice.
+ *
+ * Gets the type of change this is.
+ *
+ * Returns: the type of change.
+ *
+ * Since: 1.5
+ */
+SnapdNoticeType
+snapd_notice_get_notice_type (SnapdNotice *self)
+{
+    g_return_val_if_fail (SNAPD_IS_NOTICE (self), SNAPD_NOTICE_TYPE_UNKNOWN);
+    return self->type;
+}
+
+/**
+ * snapd_notice_get_key:
+ * @notice: a #SnapdNotice.
+ *
+ * Get the change-id or the instance-name, depending on the type.
+ *
+ * Returns: (transfer none): a string with the key.
+ *
+ * Since: 1.5
+ */
+const gchar *
+snapd_notice_get_key (SnapdNotice *self)
+{
+    g_return_val_if_fail (SNAPD_IS_NOTICE (self), NULL);
+    return self->key;
+}
+
+/**
+ * snapd_notice_get_data:
```

Comment:
Change to snapd_notice_get_last_data as this matches the field returned by snapd. We always use the snapd names as this will avoid any naming collision issues if snapd later adds a new field.

Reply:
You are right. Fixed.